### DISCUSSION TITLE
Exception has occurred: AuthorizationError  ('Client authentication failed', 401)

### DISCUSSION THREAD

**Discussion by rashedoz** [saleweaver/python-amazon-sp-api#362] [Q&A] [answered]
I keep getting this exception:
```
Exception has occurred: AuthorizationError: 
('invalid_client', 'Client authentication failed', 401)
```

I have created all the IAM User, Role, Policy, and App as per the documentation. And got all the required credentials. But still getting this error. I am really lost on how to debug this.

All the steps were followed to grant access to the IAM role with Seller central app. I also tried with just user ARN instead of user Role ARN. But same. 

The only confusion I have is that the Developer central app status is shown in 'Draft' because it's for private purposes I don't want to publish the app. When in Draft status won't it work? 

I am really lost why i am getting `sp_api.auth.exceptions.AuthorizationError: ('invalid_client', 'Client authentication failed', 401)`

**Comment by saleweaver** [comment, marked as answer]
Try to remove the `role_arn` value from the credentials. 

Did you check the duplicates to your question in discussions? There are many solutions to different problems already.

**Reply by rashedoz** [reply]
Sorry for the duplicate issue. I checked the discussion later. I found the problem after browsing the discussions. The **lwa_app_id** field was the culprit. In the Seller central APP, there was an app id. I was confusing it with the **LWA_APP_ID**.
The LWA client id and secret are together in the LWA credentials. I was following another documentation where lwa client secret was refered to as only client secret which started the confusion probably. 

Anyways thanks. I saw a few others are having the same problem most probably the same confusion.

**Reply by saleweaver** [reply]
Good you got it working. 😄
